surveySimPP/modules/PPDetectionEfficiency.py

```diff
@@ -4,15 +4,19 @@
 import sys
 import logging
 import numpy as np
+import time
 
 # Author: Grigori Fedorets
 
-def PPDetectionEfficiency(padain, threshold):
+default_rng = np.random.default_rng(int(time.time()))
```

Comment:
Why are there these call outside the function? This is now a global variable I believe which is being set multiple times.

Reply:
We have a seed already set in surveySImPP so you should be able to send default_rng through, so why are these calls needed?

Reply:
They're only a failsafe in case the functions are called without the RNG as an argument: this is how it's done already in PPRandomizeMeasurements.py so I was copying that for standardisation. I will happily remove these calls if you prefer.

Reply:
yeah, that's making it a a global variable which is bad... means you can change it anywhere and can be called anywhere

https://www.programiz.com/python-programming/global-local-nonlocal-variables

if the function is called without the RNG argument it should crash because the arguments sent to the function are incorrect (see https://stackoverflow.com/questions/38364162/in-python-how-do-you-denote-required-parameters-and-optional-parameters-in-code for an explanation)

Reply:
Okay, no problem. I'll also remove the global from PPRandomizeMeasurements.py then.

Reply:
Thanks. Otherwise I think every time the code is imported a new seed is generated, which we don't want. 

No global variables... (no wire hangers...)